forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/api/openbanking/payment/v3_1_4/domesticpayments/DomesticPaymentsApiController.java

```diff
@@ -144,32 +151,27 @@ public ResponseEntity getDomesticPaymentsDomesticPaymentId(
             String xCustomerUserAgent,
             HttpServletRequest request,
             Principal principal
-    ) throws OBErrorResponseException {
+    ) {
         // Optional endpoint - not implemented
-        return new ResponseEntity<OBWritePaymentDetailsResponse1>(HttpStatus.NOT_IMPLEMENTED);
+        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
     }
 
     private OBWriteDomesticResponse4 responseEntity(FRDomesticPaymentSubmission frPaymentSubmission, FRDomesticConsent frDomesticConsent) {
+        FRReadRefundAccount readRefundAccount = frDomesticConsent.getDomesticConsent().getData().getReadRefundAccount();
+        FRWriteDomesticDataInitiation initiation = frPaymentSubmission.getDomesticPayment().getData().getInitiation();
+        Optional<FRDomesticResponseDataRefund> refund = frDomesticResponseDataRefund(readRefundAccount, initiation);
 
-        OBWriteDomesticResponse4 obWriteDomesticResponse4 = new OBWriteDomesticResponse4()
+        return new OBWriteDomesticResponse4()
                 .data(new OBWriteDomesticResponse4Data()
                         .domesticPaymentId(frPaymentSubmission.getId())
                         .initiation(toOBWriteDomestic2DataInitiation(frDomesticConsent.getDomesticConsent().getData().getInitiation()))
                         .creationDateTime(frDomesticConsent.getCreated())
                         .statusUpdateDateTime(frDomesticConsent.getStatusUpdate())
                         .status(toOBWriteDomesticResponse4DataStatus(frDomesticConsent.getStatus()))
                         .consentId(frDomesticConsent.getId())
-                )
+                        .refund(refund.map(FRResponseDataRefundConverter::toOBWriteDomesticResponse4DataRefund).orElse(null)))
```

Comment:
Now setting the refund value directly within this initialisation section, rather than having the separate block below. This keeps the construction syntax consistent with all versions prior to v3.1.4.